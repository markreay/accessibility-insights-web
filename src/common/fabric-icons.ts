// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { registerIcons } from '@uifabric/styling';

export function initializeFabricIcons(): void {
    registerIcons({
        style: {
            MozOsxFontSmoothing: 'grayscale',
            WebkitFontSmoothing: 'antialiased',
            fontStyle: 'normal',
            fontWeight: 'normal',
            speak: 'none',
        },
        fontFace: {
            fontFamily: 'FabricMDL2Icons',
        },
        icons: {
            add: '\uE710',
            back: '\uE72B',
            calculatorAddition: '\uE948',
            cancel: '\uE711',
            checkBox: '\uE739',
            checkMark: '\uE73E',
            chevronDown: '\uE70D',
            chevronDownMed: '\uE972',
            chevronRight: '\uE76C',
            chevronRightMed: '\uE974',
            chevronUp: '\uE70E',
            chromeClose: '\uE8BB',
            chromeMinimize: '\uE921',
            circleRing: '\uEA3A',
            completedSolid: '\uEC61',
            copy: '\uE8C8',
            contactCard: '\uEEBD',
            delete: '\uE74D',
            edit: '\uE70F',
            export: '\uEDE1',
            feedback: '\uED15',
            fileHTML: '\uF2ED',
            gear: '\uE713',
            giftboxOpen: '\uF133',
            globalNavButton: '\uE700',
            hide2: '\uEF89',
            home: '\uE80F',
            incidentTriangle: '\uE814',
            info: '\uE946',
            ladybugSolid: '\uF44A',
            mail: '\uE715',
            medical: '\uEAD4',
            play: '\uE768',
            refresh: '\uE72C',
            rocket: '\uF3B3',
            scopeTemplate: '\uF2B0',
            search: '\uE721',
            send: '\uE724',
            skypeCheck: '\uEF80',
            statusCircleCheckMark: '\uF13E',
            statusErrorFull: '\uEB90',
            stop: '\uE71A',
            tag: '\uE8EC',
            testBeaker: '\uF3A5',
            testBeakerSolid: '\uF3A6',
            undo: '\uE7A7',
            unknown: '\uE9CE',
            view: '\uE890',
        },
    });
}
