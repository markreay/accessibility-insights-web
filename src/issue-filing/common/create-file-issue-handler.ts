// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
import { EnvironmentInfo } from 'common/environment-info-provider';
import { CreateIssueDetailsTextData } from 'common/types/create-issue-details-text-data';
import { IssueFilingServicePropertiesMap } from 'common/types/store-data/user-configuration-store';

import { IssueFilingUrlProvider } from '../types/issue-filing-service';

export const createFileIssueHandler = <Settings>(
    getUrl: IssueFilingUrlProvider<Settings>,
    getSettings: (data: IssueFilingServicePropertiesMap) => Settings,
) => {
    return async (
        browserAdapter: BrowserAdapter,
        servicePropertiesMap: IssueFilingServicePropertiesMap,
        issueData: CreateIssueDetailsTextData,
        environmentInfo: EnvironmentInfo,
    ): Promise<void> => {
        const serviceConfig = getSettings(servicePropertiesMap);

        const url = getUrl(serviceConfig, issueData, environmentInfo);
        await browserAdapter.createActiveTab(url);
    };
};
