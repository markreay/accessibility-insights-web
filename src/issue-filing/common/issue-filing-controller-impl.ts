// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { BaseStore } from 'common/base-store';
import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
import { EnvironmentInfo } from 'common/environment-info-provider';
import { CreateIssueDetailsTextData } from 'common/types/create-issue-details-text-data';
import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';

import { IssueFilingServiceProvider } from '../issue-filing-service-provider';

export type IssueFilingController = {
    fileIssue: (serviceKey: string, issueData: CreateIssueDetailsTextData) => void;
};

export class IssueFilingControllerImpl implements IssueFilingController {
    constructor(
        private readonly provider: IssueFilingServiceProvider,
        private readonly browserAdapter: BrowserAdapter,
        private readonly environmentInfo: EnvironmentInfo,
        private readonly userConfigurationStore: BaseStore<UserConfigurationStoreData>,
    ) {}

    public fileIssue = (serviceKey: string, issueData: CreateIssueDetailsTextData): Promise<void> => {
        const service = this.provider.forKey(serviceKey);
        const userConfigurationStoreData = this.userConfigurationStore.getState();

        return service.fileIssue(this.browserAdapter, userConfigurationStoreData.bugServicePropertiesMap, issueData, this.environmentInfo);
    };
}
