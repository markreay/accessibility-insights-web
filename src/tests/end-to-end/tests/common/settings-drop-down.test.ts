// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Browser } from '../../common/browser';
import { launchBrowser } from '../../common/browser-factory';
import { CommonSelectors } from '../../common/element-identifiers/common-selectors';
import { formatPageElementForSnapshot } from '../../common/element-snapshot-formatter';
import { Page } from '../../common/page-controllers/page';
import { TargetPage } from '../../common/page-controllers/target-page';
import { scanForAccessibilityIssues } from '../../common/scan-for-accessibility-issues';

describe('Settings Dropdown', () => {
    let browser: Browser;
    let targetPage: TargetPage;

    beforeAll(async () => {
        browser = await launchBrowser({ suppressFirstTimeDialog: true });
    });

    beforeEach(async () => {
        targetPage = await browser.newTargetPage();
    });

    afterEach(async () => {
        if (browser) {
            await browser.closeAllPages();
        }
    });

    afterAll(async () => {
        if (browser) {
            await browser.close();
            browser = undefined;
        }
    });

    it('content should match snapshot', async () => {
        const popupPage = await browser.newPopupPage(targetPage);
        const popupDropdownElement = await getDropdownPanelElement(popupPage);

        const detailsViewPage = await browser.newDetailsViewPage(targetPage);
        const detailsViewDropdownElement = await getDropdownPanelElement(detailsViewPage);

        expect(popupDropdownElement).toEqual(detailsViewDropdownElement);
        expect(popupDropdownElement).toMatchSnapshot();
    });

    it.each([true, false])('should pass accessibility validation in popup with highContrastMode=%s', async highContrastMode => {
        await browser.setHighContrastMode(highContrastMode);
        const popupPage = await browser.newPopupPage(targetPage);
        await popupPage.waitForHighContrastMode(highContrastMode);

        await popupPage.clickSelector(CommonSelectors.settingsGearButton);

        const results = await scanForAccessibilityIssues(popupPage, CommonSelectors.settingsDropdownMenu);
        expect(results).toHaveLength(0);
    });

    it.each([true, false])('should pass accessibility validation in details view with highContrastMode=%s', async highContrastMode => {
        await browser.setHighContrastMode(highContrastMode);
        const detailsViewPage = await browser.newDetailsViewPage(targetPage);
        await detailsViewPage.waitForHighContrastMode(highContrastMode);

        await detailsViewPage.clickSelector(CommonSelectors.settingsGearButton);

        const results = await scanForAccessibilityIssues(detailsViewPage, CommonSelectors.settingsDropdownMenu);
        expect(results).toHaveLength(0);
    });

    async function getDropdownPanelElement(page: Page): Promise<Node> {
        await page.clickSelector(CommonSelectors.settingsGearButton);
        return await formatPageElementForSnapshot(page, CommonSelectors.settingsDropdownMenu);
    }
});
