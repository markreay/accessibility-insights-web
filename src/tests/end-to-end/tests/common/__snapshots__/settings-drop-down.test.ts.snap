// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Dropdown content should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ms-ContextualMenu-container container-000"
    id="settings-dropdown-menu"
    tabindex="-1"
  >
    <div
      class="ms-FocusZone css-000 ms-ContextualMenu is-open root-000"
      data-focuszone-id="FocusZone000"
      role="presentation"
    >
      <ul
        class="ms-ContextualMenu-list is-open list-000"
        role="menu"
      >
        <li
          class="ms-ContextualMenu-item item-000"
          role="presentation"
        >
          <button
            aria-disabled="false"
            aria-posinset="1"
            aria-setsize="2"
            class="ms-ContextualMenu-link root-000"
            name="Settings"
            role="menuitem"
            tabindex="0"
          >
            <div
              class="ms-ContextualMenu-linkContent linkContent-000"
            >
              <i
                aria-hidden="true"
                class="ms-ContextualMenu-icon icon-000"
                data-icon-name="gear"
                role="presentation"
              >
                
              </i>
              <span
                class="ms-ContextualMenu-itemText label-000"
              >
                Settings
              </span>
            </div>
          </button>
        </li>
        <li
          class="ms-ContextualMenu-item preview-features-drop-down-button item-000"
          role="presentation"
        >
          <button
            aria-disabled="false"
            aria-posinset="2"
            aria-setsize="2"
            class="ms-ContextualMenu-link root-000"
            name="Preview features"
            role="menuitem"
            tabindex="-1"
          >
            <div
              class="ms-ContextualMenu-linkContent linkContent-000"
            >
              <i
                aria-hidden="true"
                class="ms-ContextualMenu-icon icon-000"
                data-icon-name="giftboxOpen"
                role="presentation"
              >
                
              </i>
              <span
                class="ms-ContextualMenu-itemText label-000"
              >
                Preview features
              </span>
            </div>
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
